xapian-core/include/xapian/cluster.h

```diff
@@ -316,10 +407,30 @@ class XAPIAN_VISIBILITY_DEFAULT RoundRobin {
     ~RoundRobin();
 
     // Implements RoundRobin clustering
-    ClusterSet cluster(MSet &mset, unsigned int k);
+    ClusterSet cluster(MSet &mset, unsigned int num_of_clusters);
 
     // Returns the description of the clusterer being used
     std::string get_description();
 };
-}
+
+/** Kmeans clusterer:
+ *  This clusterer implements the K-Means clustering algorithm
+ */
+class XAPIAN_VISIBILITY_DEFAULT KMeans {
```

Comment:
It's a difficult thing to get right, and there's no hard and fast rules. It's not about small pieces so much as self-contained pieces. The refactor goes from one good state to another (with a different API), so it's reasonable to have a commit that contains only that. Hope that's helpful, but let me know if I've just confusing things :)